### PR TITLE
added feature bam2fa

### PR DESCRIPTION
**Pull request by sproft**
new feature that allows reading BAM files as fasta or fastq. Includes new tests and documentation

### CODE REVIEW

**Comment by rrahn** [comment]
@temehi We should stash the commits into a single commit before merging. Can you do this with them?
Thanks a lot.

File: .gitignore [hunk 1]
```diff
@@ -5,3 +5,4 @@ sandbox
 !sandbox/CMakeLists.txt
 *.pyc
 *~
+.idea
```
Comment (h-2) [inline .gitignore]: whats this?

File: include/seqan/bam_io/bam_file.h [hunk 2]
```diff
@@ -233,14 +233,86 @@ readRecord(BamAlignmentRecord & record,
         readRecord(record, context, iter, static_cast<typename TagSelector<TTagList>::Base const &>(format));
 }
 
-// convient BamFile variant
+
+template <typename TIdString, typename TSeqString, typename TForwardIter,
+        typename TNameStore, typename TNameStoreCache, typename TStorageSpec>
+inline void
+        readRecord(TIdString & /*meta*/, TSeqString & /*seq*/,
```
Comment (h-2) [inline include/seqan/bam_io/bam_file.h]: strange indention

File: include/seqan/bam_io/bam_file.h [hunk 3]
```diff
@@ -233,14 +233,86 @@ readRecord(BamAlignmentRecord & record,
...
+                   TagSelector<> const & /*format*/)
+
+{
+    SEQAN_FAIL("BamFileIn: File format not specified.");
+}
+
+// when you compile with ZLIB, format is TaglistSelector instead of BAM. This function tells if it is BAM or SAM
+// this is for readRecord without quality
+template <typename TIdString, typename TSeqString, typename TForwardIter,
+        typename TNameStore, typename TNameStoreCache, typename TStorageSpec, typename TTagList>
```
Comment (h-2) [inline include/seqan/bam_io/bam_file.h]: strange indention

File: include/seqan/bam_io/bam_file.h [hunk 4]
```diff
@@ -233,14 +233,86 @@ readRecord(BamAlignmentRecord & record,
...
+    SEQAN_FAIL("BamFileIn: File format not specified.");
+}
+
+// when you compile with ZLIB, format is TaglistSelector instead of BAM. This function tells if it is BAM or SAM
+// this is for readRecord with quality
+template <typename TIdString, typename TSeqString, typename TQualString, typename TForwardIter,
+        typename TNameStore, typename TNameStoreCache, typename TStorageSpec, typename TTagList>
+inline void
+readRecord(TIdString & meta, TSeqString & seq, TQualString & qual,
+           BamIOContext<TNameStore, TNameStoreCache, TStorageSpec> & context,
```
Comment (h-2) [inline include/seqan/bam_io/bam_file.h]: each parameter in its own line please

File: include/seqan/bam_io/read_bam.h [hunk 5]
```diff
@@ -274,6 +274,161 @@ readRecord(BamAlignmentRecord & record,
...
+template <typename TIdString, typename TSeqString,
+        typename TCharIter, typename TNameStore, typename TNameStoreCache, typename TStorageSpec>
+inline int32_t
+_readRecord(TIdString & meta, TSeqString & seq, CharString & prevQName,
+            BamIOContext<TNameStore, TNameStoreCache, TStorageSpec> & context,
+            int32_t & remainingBytes,
+            TCharIter & it,
+            Bam const & /* tag */)
+{
+
```
Comment (h-2) [inline include/seqan/bam_io/read_bam.h]: remove newline

**Pull request closed without merging**